code/modules/spells/spell.dm

```diff
@@ -71,9 +71,10 @@ var/list/spells = typesof(/obj/effect/proc_holder/spell) //needed for the badmin
 		if(!(src in user.mob_spell_list))
 			return 0
 
-	if(user.z == ZLEVEL_CENTCOM && !centcom_cancast) //Certain spells are not allowed on the centcom zlevel
+	var/turf/T = get_turf(user)
+	if(T.z == ZLEVEL_CENTCOM && !centcom_cancast) //Certain spells are not allowed on the centcom zlevel
```

Comment:
casual reminder that this only applies to jaunt, phase shift and other high mobility spells not appropriate for the shuttle environment anyway